### PR TITLE
Feat/dashboard behaviour for project reset

### PR DESCRIPTION
**Pull request by joshenlim**
### What this is
- More specific use case is for when user hits the "Restart server" button in the project settings, so that at least the user knows when the project is back online

https://user-images.githubusercontent.com/19742402/164391591-65465dc5-1825-4886-b03c-9e5b2bedd33b.mov

### Implementation details
- Within `TestConnection`, discern if project is online by pinging the project's postgrest
  - Representation of whether both DB and KPS are both online
- If not online, show `ConnectingState`
  - Once back online, it'll refresh the page to re-trigger `TestConnection`
  - Reason why we're doing this instead of doing something similar as we track the building state is because
    - 1. The project may be ACTIVE_HEALTHY while the project is down (Can't use project status in the useEffect dependency array)
    - 2. BE doesn't track the project's status when triggering a server restart yet. Ideally it should update the project status to "RESTARTING" or something along those lines
- Clicking the side nav URLs will redirect back to the project home page (except project settings)

### CODE REVIEW

File: studio/components/layouts/ProjectLayout/ProjectLayout.tsx [hunk 1]
```diff
@@ -29,6 +33,8 @@ const ProjectLayout: FC<Props> = ({
   hideHeader = false,
   hideIconBar = false,
 }) => {
+  const { ui } = useStore()
+
   return (
```
Comment (phamhieu) [inline studio/components/layouts/ProjectLayout/ProjectLayout.tsx:38]: unused code

File: studio/components/layouts/ProjectLayout/ProjectLayout.tsx [hunk 2]
```diff
@@ -3,6 +3,10 @@ import { FC, ReactNode } from 'react'
 import { observer } from 'mobx-react-lite'
 import { useRouter } from 'next/router'
 import { useStore } from 'hooks'
+import { Button } from '@supabase/ui'
+
+import { get, head } from 'lib/common/fetch'
+import { API_URL } from 'lib/constants'
```
Comment (phamhieu) [inline studio/components/layouts/ProjectLayout/ProjectLayout.tsx:10]: unused code

**Comment by MildTomato** [comment]
Is "Connecting.." the right language here.

Should it not just be "Restarting..."

File: studio/components/interfaces/Functions/FunctionsListItem.tsx [hunk 3]
```diff
@@ -14,8 +14,6 @@ interface Props {
 
 const FunctionsListItem: FC<Props> = ({ function: item }) => {
   const { ui } = useStore()
-
-  console.log('_function', item)
```
Comment (MildTomato) [inline studio/components/interfaces/Functions/FunctionsListItem.tsx:18]: thanks

**Comment by joshenlim** [comment]
> Is "Connecting.." the right language here.

Oop - wrote on slack, just copy pasting here for visibility

yeah i was thinking about this - but the logic we've written doesnt necessarily discern between a "restarting" state and a "lost connection" state since the project's status doesnt update yet

we can probably look into re-wording for a more explicit state of which the project is in once the BE team manages to track the project status but figured for now, we'd cover a broader use case

**Review by MildTomato — APPROVED** [review]

**Pull request merged**